src/app/drive/components/FileViewer/FileViewer.tsx

```diff
@@ -43,36 +45,31 @@ interface FileViewerProps {
     renameItemFromKeyboard: ((item) => void) | undefined;
     removeItemFromKeyboard: ((item) => void) | undefined;
   };
+  handlersForSpecialItems?: {
+    handleUpdateProgress: (progress: number) => void;
+    handleUpdateThumbnail: (driveFile: PreviewFileItem, blob: Blob) => Promise<void>;
+  };
 }
 
 export interface FormatFileViewerProps {
   blob: Blob;
-  changeFile: (direction: string) => void;
+  file: PreviewFileItem;
+  changeFile?: (direction: 'next' | 'prev') => void;
+  setIsPreviewAvailable: (isPreviewAvailable: boolean) => void;
+  handlersForSpecialItems?: {
+    handleUpdateProgress: (progress: number) => void;
+    handleUpdateThumbnail: (driveFile: PreviewFileItem, blob: Blob) => Promise<void>;
+  };
 }
 
 const extensionsList = fileExtensionService.computeExtensionsLists(fileExtensionPreviewableGroups);
 
-function shouldNotBeRendered(fileExtensionGroup) {
+function extensionIsNotAllowed(fileExtensionGroup) {
```

Comment:
This function name is more like a boolean variable name, normally a function name starts with a verb.
For example, `checkIfExtensionIsAllowed`